kml.cc

```diff
@@ -1557,6 +1565,7 @@ void KmlFormat::kml_mt_hdr(const route_head* header)
 
   // TODO: How to handle clamped, floating, extruded, etc.?
   foreach (const Waypoint* tpt, header->waypoint_list) {
+    const auto* fs_igc = reinterpret_cast<igc_fsdata*>(tpt->fs.FsChainFind(kFsIGC));
```

Comment:
For now, yes.

Currently the only data being extracted is engine noise level, but there may be others. The single (has_igc_exts) lookup is a way to eliminate doing a lookup for every possible extension it _may_ have, because the `has_igc_exts` boolean is recorded once when the parser looks at a single record in the IGC file.

Without the `has_igc_exts` lookup, we would need to perform that lookup for _every_ possible extension, including when processing files not derived from IGC.

In essence, while it's currently an extra lookup, it avoids far more lookups (which are irrelevant to most formats) when more extensions are added.